src/main/kotlin/com/ampnet/headlesscmsservice/service/ContentServiceImpl.kt

```diff
@@ -0,0 +1,39 @@
+package com.ampnet.headlesscmsservice.service
+
+import com.ampnet.headlesscmsservice.exception.ErrorCode
+import com.ampnet.headlesscmsservice.exception.ResourceNotFoundException
+import com.ampnet.headlesscmsservice.persistence.model.Content
+import com.ampnet.headlesscmsservice.persistence.respository.ContentRepository
+import com.ampnet.headlesscmsservice.service.pojo.ContentListResponse
+import com.ampnet.headlesscmsservice.service.pojo.ContentResponse
+import com.ampnet.headlesscmsservice.service.pojo.ContentUpdateServiceRequest
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+@Service
+class ContentServiceImpl(private val contentRepository: ContentRepository) : ContentService {
+
+    @Transactional(readOnly = true)
+    override fun findByCoop(coop: String, key: String?, languge: String?): ContentListResponse {
+        val texts = contentRepository.findByCoopAndOptionalKeyAndOptionalLang(coop, key, languge)
+        if (texts.isEmpty()) throw ResourceNotFoundException(
+            ErrorCode.CMS_CONTENT_NOT_FOUND, "Text for $key in $languge for $coop not found"
+        )
+        return ContentListResponse(texts.map { ContentResponse(it) })
+    }
+
+    @Transactional
+    override fun updateContent(request: ContentUpdateServiceRequest): ContentResponse {
+        val text = contentRepository.findByCoopAndOptionalKeyAndOptionalLang(
+            request.coop, request.key, request.lang
+        ).firstOrNull() ?: run {
+            val text = contentRepository.save(
+                Content(UUID.randomUUID(), request.text, request.coop, request.key, request.lang)
+            )
+            return ContentResponse(text)
```

Comment:
If the user can also create the new content/text, add a test case for this and explain it in the docs.

Reply:
A regular user cannot create new content. It has to have PWA_COOP priviledge

Reply:
Ok, admin can update and also create a new entry?

Reply:
Yes